src/UI/Art/SpriteArtist/SpriteMapView.java

```diff
@@ -0,0 +1,292 @@
+
+	private int mapViewWidth;
+	private int mapViewHeight;
+
+	public SpriteMapView(GameInstance game)
+	{
+		mapArtist = new SpriteMapArtist(game, this);
+		unitArtist = new SpriteUnitArtist(game, this);
+		menuArtist = new SpriteMenuArtist(game, this);
+
+		myGame = game;
+		unitFacings = new HashMap<Commander, Boolean>();
+
+		// Locally store which direction each CO should be facing.
+		for(CommandingOfficers.Commander co : game.commanders)
+		{
+			setCommanderUnitFacing(co, game.gameMap);
+		}
+
+		// By default, we will show a 15x10 chunk of the map.
+		mapViewWidth = SpriteLibrary.baseSpriteSize * getDrawScale() * 15;
+		mapViewHeight = SpriteLibrary.baseSpriteSize * getDrawScale() * 10;
+		setPreferredSize(new Dimension(mapViewWidth, mapViewHeight));
+	}
+
+	@Override
+	public int getTileSize()
+	{
+		return SpriteLibrary.baseSpriteSize * getDrawScale();
+	}
+
+	@Override
+	public int getViewWidth()
+	{
+		return mapViewWidth;
+	}
+
+	@Override
+	public int getViewHeight()
+	{
+		return mapViewHeight;
+	}
+
+	/** Returns whether the commander's map units should be flipped horizontally when drawn. */
+	public boolean getFlipUnitFacing( Commander co )
+	{
+		return unitFacings.get(co);
+	}
+
+	/**
+	 * Set the facing direction of the CO based on the location of the HQ. If the
+	 * HQ is on the left side of the map, the units should face right, and vice versa.
+	 * @param co
+	 * @param map
+	 */
+	private void setCommanderUnitFacing(CommandingOfficers.Commander co, GameMap map)
+	{
+		for(int x = 0; x < map.mapWidth; ++x)
+		{
+			for(int y = 0; y < map.mapHeight; ++y)
+			{
+				if(map.getEnvironment(x, y).terrainType == Environment.Terrains.HQ &&
+						map.getLocation(x, y).getOwner() == co)
+				{
+					unitFacings.put(co, x >= map.mapWidth/2);
+				}
+			}
+		}
+	}
+
+	@Override
+	protected void paintComponent(Graphics g)
+	{
+		super.paintComponent(g);
+
+		// Draw base terrain
+		mapArtist.drawBaseTerrain(g);
+
+		// Draw units, buildings, trees, etc.
+		drawUnitsAndMapObjects(g);
+
+		// Draw Unit HP icons on top of everything, to make sure they are seen clearly.
+		unitArtist.drawUnitHPIcons(g);
+
+		// Apply any relevant map highlight.
+		mapArtist.drawHighlights(g);
+
+		// TODO: Consider moving the contemplated move inside of the action (in MapController)
+		//       to make the interface more consistent?
+		// Draw the movement arrow if the user is contemplating a move.
+		if(mapController.getContemplatedMove() != null)
+		{
+			mapArtist.drawMovePath(g, mapController.getContemplatedMove());
+		}
+		// Draw the movement arrow if the user is contemplating an action (but not once the action commences).
+		if(null != currentAction && null != currentAction.getMovePath() && null == currentAnimation)
+		{
+			mapArtist.drawMovePath(g, currentAction.getMovePath());
+		}
+		// Draw the acting unit so it's on top of everything.
+		if(null != currentAction) // && currentAnimation == null) // If the unit should animate when acting.
```

Comment:
Might this result in the unit being visibly drawn twice?
The comment seems to be out of date/confusing as well.